user/views.py

```diff
@@ -50,10 +51,12 @@ def privacy():
 @login_required
 def privacy_delete_trip(date):
     user = User.objects.get(id=current_user.id)
-    user.locations.remove(user.locations.get(date=datetime.fromtimestamp(
-                                             date)))
-    user.save()
-    return redirect('privacy'), 204
+    try:
+        user.locations.remove(user.locations.get(date=datetime.fromtimestamp(date)))
+        user.save()
+        return redirect(url_for('privacy'))
+    except DoesNotExist:
```

Comment:
I think you can do `User.DoesNotExist` instead of importing `DoesNotExist` from mongoengine.errors

Reply:
I confirm that it is a `mongoengine.errors.DoesNotExist`that raised because the `get` is done on `user.locations` and not on `User`.

Reply:
What about `UserLocation.DoesNotExist`? Which is more explicit about the nature of the item not found.

Reply:
I have just tried, the class `UserLocation`doesn't have a `DoesNotExist` attribute.

Reply:
Could be as it is an `EmbeddedDocument`…
Nevermind then, all good :+1: